benchmarks/src/main/java/org/elasticsearch/benchmark/xcontent/FilterContentBenchmark.java

```diff
@@ -0,0 +1,182 @@
+                Set<String> wildcardField = new HashSet<>(Arrays.asList("*stats"));
+                filters = wildcardField;
+            }
+                break;
+            case "10_wildcard_field": {
+                filters = new HashSet<>(
+                    Arrays.asList(
+                        "*stats.nodes*",
+                        "*stats.ind*",
+                        "*sta*.shards",
+                        "*stats*.xpack",
+                        "*stats.*.segments",
+                        "*stat*.*.data*",
+                        inclusive ? "*stats.**.request_cache" : "*stats.*.request_cache",
+                        inclusive ? "*stats.**.stat" : "*stats.*.stat",
+                        inclusive ? "*stats.**.threads" : "*stats.*.threads",
+                        "*source_node.t*"
+                    )
+                );
+            }
+                break;
+            default:
+                throw new IllegalArgumentException("Unknown type [" + type + "]");
+        }
+
+        Set<String> includes;
+        Set<String> excludes;
+        if (inclusive) {
+            includes = filters;
+            excludes = null;
+        } else {
+            includes = null;
+            excludes = filters;
+        }
+        parserConfig = XContentParserConfiguration.EMPTY.withFiltering(includes, excludes);
+    }
+
+    @Benchmark
+    public BytesReference filterWithParserConfigCreated() throws IOException {
+        try (BytesStreamOutput os = new BytesStreamOutput()) {
+            XContentBuilder builder = new XContentBuilder(XContentType.JSON.xContent(), os);
+            try (XContentParser parser = XContentType.JSON.xContent().createParser(parserConfig, source.streamInput())) {
+                builder.copyCurrentStructure(parser);
+                return BytesReference.bytes(builder);
+            } catch (Exception e) {
+                return new BytesArray("");
+            }
+        }
+    }
+
+    @Benchmark
+    public BytesReference filterWithoutParserConfigCreated() throws IOException {
+        try (BytesStreamOutput os = new BytesStreamOutput()) {
+            XContentBuilder builder = new XContentBuilder(XContentType.JSON.xContent(), os);
+            Set<String> includes;
+            Set<String> excludes;
+            if (inclusive) {
+                includes = filters;
+                excludes = null;
+            } else {
+                includes = null;
+                excludes = filters;
+            }
+            XContentParserConfiguration contentParserConfiguration = XContentParserConfiguration.EMPTY.withFiltering(includes, excludes);
+            try (XContentParser parser = XContentType.JSON.xContent().createParser(contentParserConfiguration, source.streamInput())) {
+                builder.copyCurrentStructure(parser);
+                return BytesReference.bytes(builder);
+            } catch (Exception e) {
+                return new BytesArray("");
```

Comment:
Could you let this bubble out? That'd make it abort the benchmark and I think we want that.

Reply:
because of the bug in https://github.com/elastic/elasticsearch/issues/80142. some tests will failed